### PR TITLE
Use user defined prompt colour variables.

### PR DESCRIPTION
**Pull request by lordlycastle**
Changes:
- Using colour variables where possible.
- Also changed the dart fish symbol slightly. From `⋊>` to `⨉⪧`, now it works with other fonts e.g. monofer.
- Added an operator at the end of the prompt.

@bucaran Any feedback? Best to try it out to see the results. One problem I encountered was that one of the colour variables wasn't defined so it gave an error when setting that colour. Should we look into this? In my case it was `fish_color_host`. The error was `set_color: Unknown color ''` perhaps just include it in the README.

Since we're not gonna use the custom colours, should I remove the `set_color_custom.fish`?

### CODE REVIEW

File: functions/fish_prompt.fish [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
-    if test ! -z "$pwd_info[3]"
-        echo -sn (set_color cff) "/$pwd_info[3]" (set_color normal)
-    end
-
+    echo -sn (prompt_pwd)
+    
     if set -l branch_name (git_branch_name)
         set -l git_glyph " on "
         set -l branch_glyph
+        # Using custom color since there isn't a defined vairbles for git stuff
```
Comment (bucaran) [inline functions/fish_prompt.fish]: Typo

**Comment by bucaran** [comment]
I'll test it shortly, in the meantime, we also need to update the bundle 'fishfile' to remove unused deps.

**Comment by lordlycastle** [comment]
> we also need to update the bundle 'fishfile' to remove unused deps.

Just deleted `pwd_info`. Also I edited my original post, you might have missed a few things.

**Comment by bucaran** [comment]
>  One problem I encountered was that one of the colour variables wasn't defined

In that case, just use double quotes around the variable. But make sure to refer to the element at index 1, since fish_colors_* are arrays sometimes.

```"$fish_color_host[1]"```

**Comment by lordlycastle** [comment]
> In that case, just use double quotes around the variable.

Still get the error...
> But make sure to refer to the element at index 1, since fish_colors_* are arrays sometimes.

Then users won't be able to add bold, underlined to their custom colours.  Kinda what we're trying to get rid of.

**Comment by bucaran** [comment]
@lordlycastle You are right. So, check whether the variable exists before:

```
set -l dartfish_color_host 555

if set -q fish_color_host
    set dartfish_color_host $fish_color_host
end

**Comment by lordlycastle** [comment]
If we check for one shouldn't we do it for all?

**Comment by bucaran** [comment]
> If we check for one shouldn't we do it for all?

Yes, I am afraid so. But only for the ones we use, so that's like 4 or 5 and that goes at the top of the function anyway.

**Comment by lordlycastle** [comment]
@bucaran Try it now! I'm setting all the undefined colour to `normal` rather than separate ones.

**Comment by bucaran** [comment]
@lordlycastle Ok, just push your new changes and I'll try it as soon as I am done with the 1.0 release for fin.

**Pull request merged**